sky/utils/command_runner.py

```diff
@@ -79,7 +82,12 @@ def ssh_options_list(ssh_private_key: Optional[str],
     }
     # SSH Control will have a severe delay when using docker_ssh_proxy_command.
     # TODO(tian): Investigate why.
-    if ssh_control_name is not None and docker_ssh_proxy_command is None:
+    # k8s instances are accessed with an ssh session using Proxycommand. The
+    # process running Proxycommand is kept running as long as the ssh session
+    # is running and the ControlMaster keeps the session, which results in
+    # 'ControlPersist' number of seconds delay per ssh commands ran.
```

Comment:
```suggestion
    # We also do not use ControlMaster when we use `kubectl port-forward` to access  Kubernetes pods over SSH+Proxycommand. This is because the
    # process running ProxyCommand is kept running as long as the ssh session
    # is running and the ControlMaster keeps the session, which results in
    # 'ControlPersist' number of seconds delay per ssh commands ran.
```